source/extensions/filters/http/grpc_streaming/config.cc

```diff
@@ -0,0 +1,133 @@
+#include "extensions/filters/http/grpc_streaming/config.h"
+
+#include "envoy/http/filter.h"
+#include "envoy/registry/registry.h"
+
+#include "common/grpc/common.h"
+#include "common/grpc/context_impl.h"
+
+#include "extensions/filters/http/grpc_streaming/message_counter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace GrpcStreaming {
+
+namespace {
+
+class GrpcStreamingFilter : public Http::StreamFilter {
+public:
+  explicit GrpcStreamingFilter(Grpc::Context& context) : context_(context) {}
+
+  // Http::StreamFilterBase
+  void onDestroy() override {}
+
+  // Http::StreamDecoderFilter
+  Http::FilterHeadersStatus decodeHeaders(Http::HeaderMap& headers, bool) override {
+    grpc_request_ = Grpc::Common::hasGrpcContentType(headers);
+    if (grpc_request_ && decoder_callbacks_) {
+      cluster_ = decoder_callbacks_->clusterInfo();
+      if (cluster_) {
+        request_names_ = context_.resolveServiceAndMethod(headers.Path());
```

Comment:
Maybe call `const_cast<const Http::HeaderMap&>(headers).Path()` to invoke `Path() const`. Otherwise the size cache will be invalidated.

Reply:
This is copied from stats code for the gRPC bridge. I'd assume a global refactor is in order?

Reply:
Let's adopt the new style in new code.
Is there a tool to do global refactor like this?

Reply:
Per comment in the associated PR, let's ignore this for now. We can deal with any potential perf issues in a follow up.

Reply:
Resolving for now.